src/gmt_init.c

```diff
@@ -2473,6 +2472,63 @@ bool gmtinit_parse_t_option (struct GMT_CTRL *GMT, char *item) {
 	return (n_errors > 0);
 }
 
+/*! Routine will decode the -wy|a|w|d|h|m|s|c<period>[/<phase>][+c<col>] arguments */
+GMT_LOCAL int gmtinit_parse_w_option (struct GMT_CTRL *GMT, char *arg) {
+
+	char *c = NULL, *s = NULL;
+
+	if (!arg || !arg[0]) return (GMT_PARSE_ERROR);	/* -w requires an argument */
+
+	if ((c = strstr (arg, "+c"))) {	/* Got a specific column */
+		if (c[2]) GMT->current.io.cycle_col = atoi (&c[2]);
+		c[0] = '\0';	/* Chop off modifier */
+		if (GMT->current.io.cycle_col < 0) {
+			GMT_Report (GMT->parent, GMT_MSG_ERROR, "Option -w: Cannot give negative (or missing) column number (%s)\n", arg);
+			c[2] = '+';	/* Restore modifier before we return */
+			return (GMT_PARSE_ERROR);			
+		}
+	}
+	else	/* Default column is the first (x) */
+		GMT->current.io.cycle_col = GMT_X;
+
+	switch (arg[0]) {	/* Look at which valid code we got */
+		case 's': GMT->current.io.cycle_operator = GMT_CYCLE_SEC; break;
+		case 'm': GMT->current.io.cycle_operator = GMT_CYCLE_MIN; break;
+		case 'h': GMT->current.io.cycle_operator = GMT_CYCLE_HOUR; break;
+		case 'd': GMT->current.io.cycle_operator = GMT_CYCLE_DAY; break;
+		case 'w': GMT->current.io.cycle_operator = GMT_CYCLE_WEEK; break;
+		case 'a': GMT->current.io.cycle_operator = GMT_CYCLE_ANNUAL; break;
+		case 'y': GMT->current.io.cycle_operator = GMT_CYCLE_YEAR; break;
+		case 'p': GMT->current.io.cycle_operator = GMT_CYCLE_CUSTOM;
```

Comment:
```suggestion
		case 'c': GMT->current.io.cycle_operator = GMT_CYCLE_CUSTOM;
```
Docs say that this is option **-wc**

Reply:
Thanks, coding changed over time as I felt c for cycle made more sense.